valve_grammar.ne

```diff
@@ -92,5 +92,6 @@ regex_flag -> [a-z]:* {% join %}
 label -> WORD | dqstring
 int -> INTEGER {% parseInt %}
 
+ALPHANUM -> [a-zA-Z0-9-_]:+ {% join %}
```

Comment:
I'm worried that this is ambiguous with `INTEGER`. In most programming languages you can't begin a symbol with a number, and I think we should adopt that rule.

```
ALPHANUM -> [a-zA-Z_][a-zA-Z0-9-_]:* {% join %}
```

Reply:
Sure! I validated that it doesn't begin with a number in the python, but this is better.

Reply:
I validated this in the python, but it would be nice to validate this in the grammar as well...

I tried to do this a different way and it didn't work (I tried a negative), and Nearley doesn't like this method either. I think you can only have one square bracket, because generating the grammar throws an error:
```
Error: Syntax error at line 39 col 22:

  ALPHANUM -> [a-zA-Z_][a-zA-Z0-9-_]
                       ^
```